src/findall.jl

```diff
@@ -65,15 +110,24 @@ julia> searchsorted_interval(Float64[], 1..3)
 1:0
 ```
 """
-function searchsorted_interval(X, i::Interval{L, R}; rev=false) where {L, R}
-    if rev === true
-        _searchsorted_begin(X, rightendpoint(i), Val(R); rev):_searchsorted_end(X, leftendpoint(i), Val(L); rev)
-    else
-        _searchsorted_begin(X, leftendpoint(i), Val(L); rev):_searchsorted_end(X, rightendpoint(i), Val(R); rev)
+function searchsorted_interval end
+
+if VERSION < v"1.11-"
+    searchsorted_interval(X, i::Interval{:closed, :closed}) = searchsortedfirst(X, leftendpoint(i))     :searchsortedlast(X, rightendpoint(i))
+    searchsorted_interval(X, i::Interval{:closed,   :open}) = searchsortedfirst(X, leftendpoint(i))     :(searchsortedfirst(X, rightendpoint(i)) - 1)
+    searchsorted_interval(X, i::Interval{  :open, :closed}) = (searchsortedlast(X, leftendpoint(i)) + 1):searchsortedlast(X, rightendpoint(i))
+    searchsorted_interval(X, i::Interval{  :open,   :open}) = (searchsortedlast(X, leftendpoint(i)) + 1):(searchsortedfirst(X, rightendpoint(i)) - 1)
```

Comment:
If we remove the keyword argument `rev`, that would be breaking. Could you add the keyword argument, or bump the version to `v0.8.0`?

Reply:
I understand the situation. If we revert #111, then we lose `rev`, and that must be considered as a breaking change. If you would like to avoid the breaking release, this PR should implement `rev` keyword argument without performance regression.

I'm okay with both choices: breaking release or adding the `rev` argument.